applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -4,3 +4,399 @@
+	if (ret != 0) {
+		LOG_INF("Unable to setup QoS: %d", ret);
+	}
+}
+
+static void stream_qos_set_cb(struct bt_audio_stream *stream)
+{
+	int ret;
+
+	ret = bt_audio_stream_enable(stream, unicast_present_nrf5340.codec.meta,
+				     unicast_present_nrf5340.codec.meta_count);
+	if (ret != 0) {
+		LOG_INF("Unable to enable stream: %d", ret);
+	}
+}
+
+static void stream_enabled_cb(struct bt_audio_stream *stream)
+{
+	int ret;
+
+	ret = bt_audio_stream_start(stream);
+	if (ret != 0) {
+		LOG_INF("Unable to start stream: %d", ret);
+	}
+}
+
+static void stream_started_cb(struct bt_audio_stream *stream)
+{
+	int ret;
+	struct event_t event;
+
+	event.event_source = EVT_SRC_LE_AUDIO;
+	event.le_audio_activity.le_audio_evt_type = LE_AUDIO_EVT_STREAMING;
+
+	ret = ctrl_events_put(&event);
+	ERR_CHK(ret);
+}
+
+static void stream_metadata_updated_cb(struct bt_audio_stream *stream)
+{
+	LOG_INF("Audio Stream %p metadata updated", (void *)stream);
+}
+
+static void stream_disabled_cb(struct bt_audio_stream *stream)
+{
+	LOG_INF("Audio Stream %p disabled", (void *)stream);
+}
+
+static void stream_stopped_cb(struct bt_audio_stream *stream)
+{
+	LOG_INF("Audio Stream %p stopped", (void *)stream);
+	int ret;
+	struct event_t event;
+
+	event.event_source = EVT_SRC_LE_AUDIO;
+	event.le_audio_activity.le_audio_evt_type = LE_AUDIO_EVT_NOT_STREAMING;
+
+	ret = ctrl_events_put(&event);
+}
+
+static void stream_released_cb(struct bt_audio_stream *stream)
+{
+	LOG_INF("Audio Stream %p released", (void *)stream);
+}
+
+static struct bt_audio_stream_ops stream_ops = {
+	.configured = stream_configured_cb,
+	.qos_set = stream_qos_set_cb,
+	.enabled = stream_enabled_cb,
+	.started = stream_started_cb,
+	.metadata_updated = stream_metadata_updated_cb,
+	.disabled = stream_disabled_cb,
+	.stopped = stream_stopped_cb,
+	.released = stream_released_cb,
+};
+
+static void add_remote_sink(struct bt_audio_ep *ep, uint8_t index)
+{
+	LOG_INF("Sink #%u: ep %p", index, (void *)ep);
+	sinks[index] = ep;
+}
+
+static void add_remote_codec(struct bt_codec *codec, int index, uint8_t type)
+{
+	if (type != BT_AUDIO_SINK && type != BT_AUDIO_SOURCE) {
+		return;
+	}
+
+	if (index < CONFIG_BT_AUDIO_UNICAST_CLIENT_PAC_COUNT) {
+		remote_codecs[index] = codec;
+	}
+}
+
+static void discover_sink_cb(struct bt_conn *conn, struct bt_codec *codec, struct bt_audio_ep *ep,
+			     struct bt_audio_discover_params *params)
+{
+	int ret;
+
+	if (params->err != 0) {
+		LOG_ERR("Discovery failed: %d", params->err);
+		return;
+	}
+
+	if (codec != NULL) {
+		add_remote_codec(codec, params->num_caps, params->type);
+		return;
+	}
+
+	if (ep != NULL) {
+		if (params->type == BT_AUDIO_SINK) {
+			add_remote_sink(ep, params->num_eps);
+		} else {
+			LOG_ERR("Invalid param type: %u", params->type);
+		}
+
+		return;
+	}
+
+	LOG_INF("Discover complete: err %d", params->err);
+
+	(void)memset(params, 0, sizeof(*params));
```

Comment:
Why do we have to set params to 0 here?

Reply:
I reference this from the upstream example.
I would assume the reason is because the param won't be cleanup  after discovery finished.
And if there's a new discovery happened in later, the old value in param which didn't been overwritten will cause issue.